R/as_draws_rvars.R

```diff
@@ -164,11 +169,24 @@ as_draws_rvars.draws_array <- function(x, ...) {
 
 #' @rdname draws_rvars
 #' @export
-as_draws_rvars.draws_df <- as_draws_rvars.draws_array
+as_draws_rvars.draws_df <- function(x, ...) {
+  x_df <- as.data.frame(x, optional = TRUE)[, variables(x, reserved = TRUE), drop = FALSE]
+  data_frame_to_matrix <- function(df) {
+    if (any(vapply(df, is.factor, logical(1)))) {
```

Comment:
shall we have a separate function/method that does the check whether any of the variables in non-numeric in a draws object?

Reply:
like, a generic function that does this for any draws type?

Reply:
(I think I'm not seeing where to use it currently, so I'm not sure exactly the signature / definition)